internal/osbuildexecutor/runner-impl-aws-ec2.go

```diff
@@ -24,6 +30,42 @@ func prepareSources(manifest []byte, store string, extraEnv []string, result boo
 	return err
 }
 
+func waitForSI(ctx context.Context, host string) bool {
+	client := http.Client{
+		Timeout: time.Second * 10,
+	}
+
+	for {
+		resp, err := client.Get(fmt.Sprintf("%s:8001/api/v1/", host))
+		if err != nil {
+			logrus.Debugf("Waiting for secure instance continues: %v", err)
+		}
+		if resp.StatusCode != 200 {
```

Comment:
It might be slightly more easier to read when checking on `if resp.StatusCode == 200 { return true}`, this way the flow is slightly more linear, maybe something like?
```go
	for {
		resp, err := client.Get(fmt.Sprintf("%s:8001/api/v1/", host))
		if err != nil {
			logrus.Debugf("Waiting for secure instance continues: %v", err)
		}
                defer resp.Body.Close()
		if resp.StatusCode == 200 {
			return true
		}
		body, err := io.ReadAll(resp.Body)
		if err != nil {
			logrus.Warningf("Unable to read body waiting for secure instance: %v", err)
		}
		logrus.Debugf("Waiting for secure instance continues: %s", body)

		select {
		case <-ctx.Done():
			logrus.Error("Timeout waiting for secure instance to spin up")
			return false
		}
	}
```